src/rgw/rgw_crypt.cc

```diff
@@ -24,6 +24,19 @@
 #include <cryptopp/cryptlib.h>
 #include <cryptopp/modes.h>
 #include <cryptopp/aes.h>
+
+/*
+ * patching for https://github.com/openssl/openssl/issues/1437
+ * newer versions of civetweb do not experience problems
+ */
+extern "C"
+{
+bool SSL_CTX_set_ecdh_auto(void* dummy, int onoff)
```

Comment:
@yehudasa I do not feel comfortable answering question about updating civetweb. Do we know if we can safely update civetweb no newer version?
@tchaikov I was not aware that this problem affects more then radosgw. In that case this code can be moved to some generic ceph parts like ceph_crypto.cc